web/src/components/storage/VolumeForm.jsx

```diff
@@ -81,6 +79,224 @@ const MountPointFormSelect = ({ volumes, ...formSelectProps }) => {
   );
 };
 
+/**
+ * Btrfs file system type can be offered with two flavors, with and without snapshots.
+ */
+const BTRFS_WITH_SNAPSHOTS = "BtrfsWithSnapshots";
+
+/**
+ * Possible file system type options for a volume.
+ * @function
+ *
+ * @param {import(~/clients/storage).Volume} volume
+ * @returns {string[]}
+ */
+const fsOptions = (volume) => {
+  const options = (volume, fsType) => {
+    if (fsType !== "Btrfs") return fsType;
+
+    const { snapshotsConfigurable } = volume.outline;
+    if (snapshotsConfigurable) return [BTRFS_WITH_SNAPSHOTS, fsType];
+
+    return (volume.snapshots ? BTRFS_WITH_SNAPSHOTS : fsType);
+  };
+
+  return volume.outline.fsTypes.flatMap(fsType => options(volume, fsType));
+};
+
+/**
+ * File system option according to the given type and the value of snapshots.
+ * @function
+ *
+ * @param {FsValue} value
+ * @returns {string}
+ */
+const fsOption = ({ fsType, snapshots }) => {
+  if (fsType !== "Btrfs") return fsType;
+
+  return (snapshots ? BTRFS_WITH_SNAPSHOTS : fsType);
+};
+
+/**
+ * Label for a file system type option.
+ * @function
+ *
+ * @param {string} fsOption
+ * @returns {string}
+ */
+const fsOptionLabel = (fsOption) => {
+  return (fsOption === BTRFS_WITH_SNAPSHOTS ? sprintf("Btrfs %s", _("with snapshots")) : fsOption);
+};
+
+/**
+ * File system properties from a file system type option.
+ * @function
+ *
+ * @param {string} fsOption
+ * @returns {FsValue}
+ */
+const fsValue = (fsOption) => {
+  if (fsOption === BTRFS_WITH_SNAPSHOTS)
+    return { fsType: "Btrfs", snapshots: true };
+  else
+    return { fsType: fsOption, snapshots: false };
+};
+
+/**
+ * Option for selecting a file system type.
+ * @component
+ *
+ * @param {object} props
+ * @param {string} props.fsOption - File system type option.
+ *
+ * @returns {ReactComponent}
+ */
+const FsSelectOption = ({ fsOption }) => {
+  const label = () => {
+    if (fsOption === BTRFS_WITH_SNAPSHOTS) return "Btrfs";
+    return fsOption;
+  };
+
+  const details = () => {
+    if (fsOption === BTRFS_WITH_SNAPSHOTS) return _("with snapshots");
+    return undefined;
+  };
+
+  const description = () => {
+    if (fsOption === BTRFS_WITH_SNAPSHOTS)
+      // TRANSLATORS: description of Btrfs snapshots feature.
+      return _("Allows rolling back any change done to the system and restoring its previous state");
```

Comment:
The description looks good. Could we have something similar for the other file systems?

At least for Ext2 we should mention that it does not support journaling so it can possibly get corrupted on crash or power outage...

Reply:
I think it makes sense to add some key info to other file system types too. That could help users to select the proper type for his/her use case. But, if you don't mind, I would address it in a seperate PR. We need to find good helps for each type.